connectors/sources/google_cloud_storage.py

```diff
@@ -199,8 +185,54 @@ async def _api_call(
                 )
                 await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry_counter)
 
+    def _validate_configurations(self):
+        """Validates whether user input is empty or not for configuration field.
+
+        Raises:
+            Exception: Configured keys can't be empty.
+        """
+        if self.configuration["service_account_credentials"] == "":
+            raise Exception("service_account_credentials can't be empty.")
+
+    def get_pem_format(self, private_key, max_split=-1):
+        """Convert key into PEM format.
+
+        Args:
+            private_key (str): Private_key in raw format.
+            max_split (int): Specifies how many splits to do. Defaults to -1.
+
+        Returns:
+            string: PEM format
+        """
+        private_key = private_key.replace(" ", "\n")
+        private_key = " ".join(private_key.split("\n", max_split))
+        private_key = " ".join(private_key.rsplit("\n", max_split))
+        return private_key
+
+    def _initialize_configurations(self):
+        """Initialize the ServiceAccountCreds"""
+        json_creds = json.loads(self.configuration["service_account_credentials"])
+        if (
+            not RUNNING_FTEST
+            and json_creds.get("private_key")
+            and "\n" not in json_creds["private_key"]
+        ):
+            json_creds["private_key"] = self.get_pem_format(
+                private_key=json_creds["private_key"].strip(), max_split=2
+            )
+
+        self.service_account_credentials = ServiceAccountCreds(
+            scopes=[CLOUD_STORAGE_READ_ONLY_SCOPE],
+            **json_creds,
+        )
+        self.user_project_id = self.service_account_credentials.project_id
+
     async def ping(self):
-        """Verify the connection with Google Cloud Storage"""
+        """Verify the configurations and connection with Google Cloud Storage"""
+
+        self._validate_configurations()
+        self._initialize_configurations()
+
         if RUNNING_FTEST:
```

Comment:
Same as above. Do we really need a check for `RUNNING_FTEST` here?

Reply:
The change was added in this PR #264 by @tarekziade. The GCS E2E takes care of the ping also, but I suppose it was done to reduce the total execution time.